Orange/evaluation/testing.py

```diff
@@ -447,12 +448,20 @@ def __init__(self, data, learners, n_resamples=10, train_size=None,
         self.n_resamples = n_resamples
         self.train_size = train_size
         self.test_size = test_size
+        self.stratified = stratified
         self.random_state = random_state
 
-        indices = skl_cross_validation.ShuffleSplit(
-            len(data), n_iter=self.n_resamples, train_size=self.train_size,
-            test_size=test_size, random_state=self.random_state
-        )
+        if self.stratified and data.domain.has_discrete_class:
+            Y = data.Y.copy().flatten()
```

Comment:
Ah, yes, no longer the case after https://github.com/biolab/orange3/pull/1004. I don't think a copy is needed. Otherwise, the PR looks great! I suggest you start working on your proposal. :smiley: